src/pages/stacks-blockchain/integrate-stacking.md

```diff
@@ -0,0 +1,366 @@
+
+-> Stacking execution will differ between mainnet and testnet in terms of cycle times and participation thresholds
+
+With the obtained PoX info, you can present if Stacking is executed in the next cycle, when the next cycle begins, the duration of a cycle, and the minimum micro-Stacks required to participate:
+
+```js
+// will Stacking be executed in the next cycle?
+const stackingExecution = poxInfo.rejection_votes_left_required > 0;
+
+// how long (in seconds) is a Stacking cycle?
+const cycleDuration = poxInfo.reward_cycle_length * blocktimeInfo.testnet.target_block_time;
+
+// how much time is left (in seconds) until the next cycle begins?
+const secondsToNextCycle =
+  (poxInfo.reward_cycle_length -
+    ((coreInfo.burn_block_height - poxInfo.first_burnchain_block_height) %
+      poxInfo.reward_cycle_length)) *
+  blocktimeInfo.testnet.target_block_time;
+
+// the actual datetime of the next cycle start
+const nextCycleStartingAt = new Date();
+nextCycleStartingAt.setSeconds(nextCycleStartingAt.getSeconds() + secondsToNextCycle);
+
+console.log({
+  stackingExecution,
+  cycleDuration,
+  nextCycleStartingAt,
+  // mimnimum micro-Stacks required to participate
+  minimumUSTX: poxInfo.min_amount_ustx,
+});
+```
+
+Users need to have sufficient Stacks (STX) tokens to participate in Stacking. With the Stacking info, this can be verified easily:
+
+```js
+const accounts = new AccountsApi(apiConfig);
+
+const accountBalance = await accounts.getAccountBalance({
+  principal: stxAddress,
+});
+
+const accountSTXBalance = accountBalance.stx.balance;
+
+// enough balance for participation?
+const canParticipate = accountSTXBalance >= poxInfo.min_amount_ustx;
+
+console.log({
+  stxAddress,
+  accountSTXBalance,
+  canParticipate,
+});
+```
+
+For testing purposes, you can use a faucet to obtain STX tokens:
+
+```shell
+curl -XPOST "https://stacks-node-api.blockstack.org/extended/v1/faucets/stx?address=<stxAddress>&stacking=true"
+```
+
+You will have to wait a few minutes for the transaction to complete.
+
+Users can select how many cycles they would like to participate in. To help with that decision, the unlocking time can be estimated:
+
+```js
+// this would be provided by the user
+let numberOfCycles = 3;
+
+// the projected datetime for the unlocking of tokens
+const unlockingAt = new Date(nextCycleStartingAt);
+unlockingAt.setSeconds(
+  unlockingAt.getSeconds() +
+    poxInfo.reward_cycle_length * numberOfCycles * blocktimeInfo.testnet.target_block_time
+);
+```
+
+## Step 4: Verify stacking eligibility
+
+At this point, your app shows Stacking details. If Stacking is executed and the user has enough funds, the user should be asked to provide input for the amount of micro-STX to lockup and a bitcoin address to be used to pay out rewards.
```

Comment:
We should be consistent with the units, microstacks instead of "micro-STX"?

Reply:
I replaced with micro-Stacks